From e90fdd56704d04af882b7f11a8bc022e2d615d37 Mon Sep 17 00:00:00 2001
From: Mario Limonciello <superm1@gmail.com>
Date: Fri, 24 Nov 2023 20:37:01 -0600
Subject: [PATCH] amd-gpu: Fix probing mistake

While trying to fix a casting error, I accidentally broke 64 bit return
addresses entirely.

Fixes: 5bb7ab869 ("trivial: amd-gpu: fix a casting error on 32 bit x86")
---
 plugins/amd-gpu/fu-amd-gpu-device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/amd-gpu/fu-amd-gpu-device.c b/plugins/amd-gpu/fu-amd-gpu-device.c
index 17db1b024..547c33e18 100644
--- a/plugins/amd-gpu/fu-amd-gpu-device.c
+++ b/plugins/amd-gpu/fu-amd-gpu-device.c
@@ -111,7 +111,7 @@ fu_amd_gpu_device_setup(FuDevice *device, GError **error)
 	struct drm_amdgpu_info_vbios vbios_info;
 	struct drm_amdgpu_info request = {
 	    .query = AMDGPU_INFO_VBIOS,
-	    .return_pointer = GPOINTER_TO_UINT(&vbios_info),
+	    .return_pointer = GPOINTER_TO_SIZE(&vbios_info),
 	    .return_size = sizeof(struct drm_amdgpu_info_vbios),
 	    .vbios_info.type = AMDGPU_INFO_VBIOS_INFO,
 	};
-- 
2.34.1

